pkgs/os-specific/linux/nvidia-x11/generic.nix

```diff
@@ -43,8 +43,20 @@ let
       else throw "nvidia-x11 does not support platform ${stdenv.system}";
 
     # patch to get the nvidia and nvidiaBeta driver to compile on kernel 4.10
-    patches = if libsOnly || versionOlder version "375"
+    patches = if libsOnly
               then null
+              else if versionOlder version "340"
+              then null
+              else if versionOlder version "375"
+              then [
+                     (fetchurl {
+                         url = https://git.archlinux.org/svntogit/packages.git/plain/trunk/4.10.0_kernel.patch?h=packages/nvidia-340xx;
+                         sha256 = "08k2phr9kawg6a3v88d4zkj7gdlih29gm5a1gmhpgmvd926k0z5l";
+                     })
+                         # from https://git.archlinux.org/svntogit/packages.git/plain/trunk/fs52243.patch?h=packages/nvidia-340xx
+                         # with datestamps removed
+                     ./fs52243.patch
+                   ]
```

Comment:
The patch failed to apply otherwise - it couldn't find the matching file because nixos resets datestamps.